bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/AbstractTextEditor.java

```diff
@@ -7095,6 +7114,38 @@ protected void doRestoreState(IMemento memento) {
 			textWidget.setHorizontalPixel(horizontalPixel.intValue());
 	}
 
+	@SuppressWarnings("boxing")
+	private ISelection readSelectionFromMemento(IMemento memento) {
+		if (memento.getBoolean(TAG_IS_MULTISELECTION)) {
+			int selectionRegions = memento.getInteger(TAG_MULTISELECTION_COUNT);
+			Vector<IRegion> regions = new Vector<>();
```

Comment:
Since the only operation here is appending content, using `LinkedList` might be more suitable.